File: Data Structures & Algorithms/duplicate-integer/submission-18.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        let hash = {}
        for (const num of nums) {
            if (!hash[num]) {
                hash[num] = true
            } else {
                return true
            }
        }

        return false
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-18.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = {}
        for i, num in enumerate(nums):
            comp = target - num

            if comp not in d:
                d[num] = i
            else:
                return [d[comp], i]
        return []